src/server/pfs/drive/driver.go

```diff
@@ -848,19 +848,22 @@ func (d *driver) inspectFile(file *pfs.File, filterShard *pfs.Shard, shard uint6
 					}
 
 					if !children[child] && !deletedChildren[child] {
-						fileInfo.Children = append(
-							fileInfo.Children,
-							client.NewFile(commit.Repo.Name, commit.ID, child),
-						)
-						if recurse {
-							childFileInfo, _, err := d.inspectFile(&pfs.File{
-								Commit: file.Commit,
-								Path:   child,
-							}, filterShard, shard, from, recurse, unsafe)
-							if err != nil {
-								return nil, nil, err
+						childFile := client.NewFile(commit.Repo.Name, commit.ID, child)
+						if pfsserver.FileInShard(filterShard, childFile) {
+							fileInfo.Children = append(
+								fileInfo.Children,
+								client.NewFile(commit.Repo.Name, commit.ID, child),
+							)
+							if recurse {
+								childFileInfo, _, err := d.inspectFile(&pfs.File{
+									Commit: file.Commit,
+									Path:   child,
+								}, filterShard, shard, from, recurse, unsafe)
+								if err != nil {
+									return nil, nil, err
+								}
+								fileInfo.SizeBytes += childFileInfo.SizeBytes
```

Comment:
What's this change doing? Can't quite grok it, were we just not checking the filterShard when we should have been?